plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -406,7 +406,10 @@ export const createSessionReplayEvent = (
 ) => {
     const timestamps = events
         .filter((e) => !!e?.timestamp)
-        .map((e) => castTimestampOrNow(DateTime.fromMillis(e.timestamp), TimestampFormat.ClickHouse))
+        .filter((e) => Math.sign(e.timestamp) === 1)
```

Comment:
I wonder if these two are unnecessary with the the validation below? Two fewer operations in a relatively hot path (which assumes `e.isValid` is cheap 🙈)

Reply:
They are not unnecessary!